File: JavaScript/198-House Robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    if (nums.length === 0) return 0;
    
    let prev = 0, curr = nums[0]
    
    for (let i=1; i<nums.length; i++){
        [prev, curr] = [curr, Math.max(curr, nums[i]+prev)];
    }
    
    return curr;
    
};

File: C++/198-House Robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int N = nums.size();
        vector<int> res(N+2, 0);
        
        for (int i=0; i<N; i++) {
            res[i+2] = max(res[i+1], res[i] + nums[i]);
        }
        
        return res[N+1];
    }
};